github/repos.go

```diff
@@ -735,6 +738,12 @@ type ProtectionRequest struct {
 	RequiredPullRequestReviews *PullRequestReviewsEnforcementRequest `json:"required_pull_request_reviews"`
 	EnforceAdmins              bool                                  `json:"enforce_admins"`
 	Restrictions               *BranchRestrictionsRequest            `json:"restrictions"`
+	// Enforces a linear commit Git history, which prevents anyone from pushing merge commits to a branch.
+	RequireLinearHistory *bool `json:"required_linear_history,omiteempty"`
+	// Permits force pushes to the protected branch by anyone with write access to the repository
```

Comment:
Every new comment in this PR needs to be a complete sentence ending in a period (".") so that the auto-generated Godocs will read nicely, and to be consistent with the rest of the repo.  Please check each comment and fix as needed.